pandas/tests/indexes/timedeltas/test_construction.py

```diff
@@ -128,10 +128,11 @@ def test_constructor_coverage(self):
 
         msg = 'periods must be a number, got foo'
         with pytest.raises(TypeError, match=msg):
-            TimedeltaIndex(start='1 days', periods='foo', freq='D')
+            timedelta_range(start='1 days', periods='foo', freq='D')
 
-        pytest.raises(ValueError, TimedeltaIndex, start='1 days',
-                      end='10 days')
+        with pytest.raises(ValueError):
```

Comment:
I'm neutral about enforcing it, but if we want to do it I think we can easily find a regex for it, and we can add the check in a single line.

Do you want to create a issue yourself with the tag "needs discussion" and if nobody is opposed, we convert it to "good first time issue"? I can do it myself if you prefer. Not sure how many cases we have to replace, but I'm sure there will be people happy to help.

Reply:
@datapythonista: Sure thing.  I cc'ed you since you have been doing a lot of work with regards to style (albeit doc-focused).

Wrote it up in #23922.